flawfinder

```diff
@@ -418,9 +418,12 @@ class Hit(object):
     # Show as CSV format
     def show_csv(self):
         csv_writer.writerow([
-            self.filename, self.line, self.column, self.level, self.category,
-            self.name, self.warning, self.suggestion, self.note,
-            self.cwes(), self.context_text, self.fingerprint()
+            self.filename.encode('utf8'), self.line,
```

Comment:
Filenames don't have an encoding. They're a sequence of bytes, that might not conform to any encoding system. Python3 has a hack in its string handling to handle arbitrary byte sequence to deal with this, because filenames aren't strings, but the ".encode" might interfere with that.